### ISSUE TITLE
epic: Multi-level Authorization for pod access

### ISSUE THREAD

**Issue by davidfestal** [kcp-dev/kcp#1972]
### TL;DR

Block unauthorized access to physical cluster Pods from KCP, through a multi-level authorization implementation.  

### Progress tracking <a href="#progress-tracking" id="progress-tracking">#</a>

EPIC detailed issues and overall progress can be tracked with the following [project view](https://github.com/orgs/kcp-dev/projects/1/views/38)

### Demo Objective

- [x] No unauthorized user can in nay way have access to either the internal service URL, or to the Syncer endpoint.

### Action Items

- [X] Scope of the current demo is necessary to fit in the prototype boundaries
- [ ] Contribute to the final demo script and recording

### Work items

- [x] #1973
  - This might be provided as is by default KCP RBAC-based authorization, but has to be tested
- [x] #1974
- [x] #1975
